src/blame.c

```diff
@@ -210,77 +207,27 @@ get_blame_commit(struct view *view, const char *id)
 static struct blame_commit *
 read_blame_commit(struct view *view, const char *text, struct blame_state *state)
 {
-	struct blame_header header;
 	struct blame_commit *commit;
-	struct blame *blame;
 
-	if (!parse_blame_header(&header, text, view->lines))
+	if (!parse_blame_header(&state->header, text))
 		return NULL;
 
 	commit = get_blame_commit(view, text);
 	if (!commit)
 		return NULL;
 
-	state->blamed += header.group;
-	while (header.group--) {
-		struct line *line = &view->line[header.lineno + header.group - 1];
-
-		blame = line->data;
-		blame->commit = commit;
-		blame->lineno = header.orig_lineno + header.group - 1;
-		line->dirty = 1;
-	}
-
 	return commit;
 }
 
-static bool
-blame_read_file(struct view *view, struct buffer *buf, struct blame_state *state)
-{
-	if (!buf) {
-		const char *blame_argv[] = {
-			"git", "blame", encoding_arg, "%(blameargs)", "--incremental",
```

Comment:
The implementation looks solid. I'm wondering though, is it really necessary to drop `--incremental` to support `-L` and `--textconv`?

Probably not a huge loss although incremental mode can be nice.
In theory `git blame --incremental -C +123` could try to find the commit that introduced line 123 first, but that's not what happens today.